File: backend/chessapp/application/dtos/chess_game_dto.py
from dataclasses import dataclass
from datetime import datetime
from ...domain.kernel.base import BaseResponse


@dataclass(frozen=True, kw_only=True)
class GameStateDto:
    turn: str
    started: bool
    finished: bool
    check_side: str
    check_position: str
    legal_moves: str = ""

@dataclass(frozen=True, kw_only=True)
class GameFormatDto:
    value: str
    white_remaining_time: float
    black_remaining_time: float
    move_increment: float

@dataclass(frozen=True, kw_only=True)
class PlayersDto:
    white_id: str
    black_id: str

@dataclass(frozen=True, kw_only=True)
class ChessGameDto(BaseResponse):
    game_id: str
    moves_count: int
    date: datetime
    name: str
    status: str
    state: GameStateDto
    game_format: GameFormatDto
    players: PlayersDto
    board: str
    history: str

File: frontend/src/app/services/models/chess-game-dto.ts
export class ChessGameDto {
    game_id: string;
    moves_count: number;
    date: Date;
    name: string;
    state: GameStateDto;
    game_format: GameFormatDto;
    players: PlayersDto;
    board: string;
    history: string;
}

export class GameStateDto {
    turn: string;
    started: boolean;
    finished: boolean;
    check_side: string | null;
    check_position: string | null;
    legal_moves: string;
}

export class GameFormatDto {
    value: string;
    white_remaining_time: number;
    black_remaining_time: number;
    move_increment: number;
}

export class PlayersDto {
    white_id: string;
    black_id: string;
}

export class PieceDto {
    abbreviation: string;
    moved: boolean;

    public constructor(init?: Partial<PieceDto>) {
        Object.assign(this, init);
    }
}